### ISSUE TITLE
trailing_stop and custom_stoploss difference in backtesting 

### ISSUE THREAD

**Issue by ZahariaLevy** [freqtrade/freqtrade#7616]
## Describe your environment

  * Operating system: Windows 11
  * Python Version: 3.10.7
  * Freqtrade Version: 2022.9.1
  
## Your question

So I'm trying to implement `custom_stoploss` but getting very significant differences in comparison to regular `trailing_stop`. First I ran test with this trailing stop loss values:
```
    "trailing": {
      "trailing_stop": true,
      "trailing_stop_positive": 0.01,
      "trailing_stop_positive_offset": 0.05,
      "trailing_only_offset_is_reached": true
    }
```
Then I disabled it and ran the test again, but now with `use_custom_stoploss` enabled and this stop loss function:
```
    def custom_stoploss(self, pair: str, trade: 'Trade', current_time: datetime,
                        current_rate: float, current_profit: float, **kwargs) -> float:
        if current_profit >= 0.05:
            return -0.01

        return 1
```
And my profit results become four times lower. 

I'm having `can_short` set to `True` because I'm using it with binance features, so maybe this have impact. So I even tried to use `trade.is_short` value in different ways, in case `custom_stoploss` doesn't consider it by default, for example like this: 
```
    def custom_stoploss(self, pair: str, trade: 'Trade', current_time: datetime,
                        current_rate: float, current_profit: float, **kwargs) -> float:
        if current_profit >= 0.05:
            return 0.01 if trade.is_short else -0.01

        return 1 if trade.is_short else -1
```
But still it doesn't give me a result close to `trailing_stop`

**Comment by xmatthias** [comment]
assuming we're talking backtesting (live differences won't be noticably, really).

When using the builtin trailing_stop  - we can take some [assumptions](https://www.freqtrade.io/en/stable/backtesting/#assumptions-made-by-backtesting).
It'll adjust on high - and then fill on low (especially with 1% trailing, you'll quite often hit this scenario - which is most likely overstated profit - which you won't reach in live).
in reality, you should try to set your trailing stop > an average candle move (if a candle's low/high ratio is 2% - set trailing stop > 2%).

for custom stoploss, we don't take these assumptions, as we don't know what the logic actually is.

As such, custom stoploss is adjusted against the candle's "open" - price - and then evaluated against the candle's low (inverse side for shorts, obviously).

if you use `--timeframe-detail` for your backtest - you'll notice that your profit with the builtin stoploss will most likely be lower, too.

**Comment by ZahariaLevy** [comment]
Yes it's backtesting. I'm running it with `--timeframe 5m` and `--timeframe-detail 1m`

So to achieve similar result between builtin trailing_stop, custom one and with real run, I should increase my trailing value? But as I understand with trailing stop > an average candle move it will be different from pair to pair, and it's not possible to set for builtin trailing_stop?

**Comment by ZahariaLevy** [comment]
And differences in results event with:
```
"trailing": {
      "trailing_stop": true,
      "trailing_stop_positive": 0.05,
      "trailing_stop_positive_offset": 0.1,
      "trailing_only_offset_is_reached": true
    }
```
and
```
    use_custom_stoploss = True

    def custom_stoploss(self, pair: str, trade: 'Trade', current_time: datetime,
                        current_rate: float, current_profit: float, **kwargs) -> float:
        if current_profit >= 0.1:
            return -0.05

        return 1
```
are pretty large

**Comment by xmatthias** [comment]
reversing the conditions won't help you - the same limitation (high vs. open) will still apply, irrelevant if you use 1% as stop value - or as offset value.

**Comment by ZahariaLevy** [comment]
But I'm using here 10% and 5% values, not 1%

**Comment by xmatthias** [comment]
yes - but the differences (as explained above) will still be there - as adjustments will no longer happen on the candle's high - but against the candle's open.

**Comment by ZahariaLevy** [comment]
And which one should usually closer to real trading `custom_stoploss` or `trailing_stop`?

**Comment by xmatthias** [comment]
That will depend on how you use it.

using custom_stoploss to 1:1 simulate trailing-stop is basically pointless - so trailing_stop will be more precise in this case.
the real usecase for custom_stoploss is to have more complex / advanced logic - where you can no longer compare the scnearios, as trailing-stop can't fully reconstruct such a scenario.
